src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/config/ChangelogMigration.kt

```diff
@@ -0,0 +1,130 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.config
+
+import mu.KLogging
+import org.springframework.context.annotation.Configuration
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.ReferralAmendmentDetails
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Changelog
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.ReferralDetails
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.AuthUserRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ChangelogRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ReferralDetailsRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.AmendTopic
+import java.time.OffsetDateTime
+import java.time.temporal.ChronoUnit
+import java.util.UUID
+import javax.annotation.PostConstruct
+
+@Configuration
+class ChangelogMigration(
+  private val referralDetailsRepository: ReferralDetailsRepository,
+  private val changelogRepository: ChangelogRepository,
+  private val authUserRepository: AuthUserRepository
+) {
+  companion object : KLogging()
+
+  @PostConstruct
```

Comment:
I like the `@PostConstruct` idea!

Unfortunately, our runtime environment requires us to make this code idempotent and concurrency-safe.

At startup, we have six instances of the application in production starting **concurrently**:
- 3x hmpps-interventions-service-api
- 2x hmpps-interventions-service-dashboard
- 1x hmpps-interventions-service-performance-report
- The NDMIS CronJob, when it starts, will also execute this.

Additionally, the pods have limited time to boot up, and Kubernetes might kill and restart them if they don't start up quickly enough, so this has to be **performant or async**.

It is probably less of a headache to make this a one-off `Job` instead of reasoning about the effects of such concurrency.